### DISCUSSION TITLE
Having issues serving static files

### DISCUSSION THREAD

**Discussion by VictorioBerra** [pocketbase/pocketbase#4955] [Q&A]
Gives me a 404 with http://127.0.0.1:8090/ and http://127.0.0.1:8090/index.html:

```js
routerAdd(
  "GET",
  "/*",
  $apis.staticDirectoryHandler(publicDir, true)
);
```

Works only if I specify `index.html` via http://127.0.0.1:8090/public/index.html

```js
routerAdd(
  "GET",
  "/public/*",
  $apis.staticDirectoryHandler(publicDir, true)
);
```

This is on WSL Ubuntu

**Comment by ganigeorgiev** [comment]
If you use the prebuilt executable there is already a `/*` route binded to the `pb_public` directory (in your case most likely your route is being overwritten by it).

In any case, @VictorioBerra this is not the first time you posted something like this. Please don't treat the discussions as LLM chat and try to elaborate on your issues as much as possible to understand the use case and what you are trying to do. Reading the above, neither the title, nor the description provide information why pb_public doesn't work in your case. See [XY problem](https://xyproblem.info/). I'll close this discussion to avoid the needless back and forth but if you need further help, please elaborate.

**Reply by VictorioBerra** [reply]
@ganigeorgiev 

That worked. I did jump the gun by going to Q/A. It should have been more obvious that the route was being overwritten and the docs mention: 

> http://127.0.0.1:8090/ - if pb_public directory exists, serves the static content from it (html, css, images, etc.)

Although I do not have a `/pb_public` folder and my route was still being clobbered:

![image](https://github.com/pocketbase/pocketbase/assets/2934507/d3c40ec9-5ad7-4f3d-944a-29df017d42f3)

Also the docs here show an example of serving static files:

https://pocketbase.io/docs/js-routing/#serving-static-files

> routerAdd("GET", "/*", $apis.staticDirectoryHandler("/path/to/public", false))

There is no mention in that section that the code will simply not work, with no error or feedback if the pb_public directory exists.
